solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -56,19 +56,11 @@ pub mod solana_ibc {
         let mut store = IbcStorage(Rc::new(RefCell::new(inner)));
         let mut router = store.clone();
 
-        let mut errors = Vec::new();
-        for msg in messages {
-            let msg = ibc_proto::google::protobuf::Any {
-                type_url: msg.type_url,
-                value: msg.value,
-            };
-            let res = ibc::core::MsgEnvelope::try_from(msg).and_then(|msg| {
-                ibc::core::dispatch(&mut store, &mut router, msg)
-            });
-            if let Err(err) = res {
-                errors.push(err);
-            }
-        }
+        let errors = messages
+            .into_iter()
+            .map(|msg| ibc::core::dispatch(&mut store, &mut router, msg))
+            .filter_map(core::result::Result::err)
+            .collect::<Vec<_>>();
 
         // Drop refcount on store so we can unwrap the Rc object below.
```

Comment:
this comment line is old right? we aren't unwrapping anything

Reply:
We are unwrapping the store so that we can log it. But we are only doing it for testing so if we dont log it, we can remove the drop.

Reply:
`Rc::try_unwrap(store.0)` is what the comment referrs to.